pkg/cri/server/container_stats_list_linux.go

```diff
@@ -157,44 +238,37 @@ func (c *criService) cpuContainerStats(ID string, isSandbox bool, stats interfac
 	return nil, nil
 }
 
-func (c *criService) memoryContainerStats(ID string, stats interface{}, timestamp time.Time) (*runtime.MemoryUsage, error) {
-	switch metrics := stats.(type) {
-	case *v1.Metrics:
-		if metrics.Memory != nil && metrics.Memory.Usage != nil {
-			workingSetBytes := getWorkingSet(metrics.Memory)
+func (c *criService) setContainerStats(containerID string, isSandbox bool, cs *containerstorestats.ContainerStats) error {
+	var oldStats *stats.ContainerStats
 
-			return &runtime.MemoryUsage{
-				Timestamp: timestamp.UnixNano(),
-				WorkingSetBytes: &runtime.UInt64Value{
-					Value: workingSetBytes,
-				},
-				AvailableBytes:  &runtime.UInt64Value{Value: getAvailableBytes(metrics.Memory, workingSetBytes)},
-				UsageBytes:      &runtime.UInt64Value{Value: metrics.Memory.Usage.Usage},
-				RssBytes:        &runtime.UInt64Value{Value: metrics.Memory.TotalRSS},
-				PageFaults:      &runtime.UInt64Value{Value: metrics.Memory.TotalPgFault},
-				MajorPageFaults: &runtime.UInt64Value{Value: metrics.Memory.TotalPgMajFault},
-			}, nil
+	if isSandbox {
+		sandbox, err := c.sandboxStore.Get(containerID)
+		if err != nil {
+			return fmt.Errorf("failed to get sandbox container: %s: %w", containerID, err)
 		}
-	case *v2.Metrics:
-		if metrics.Memory != nil {
-			workingSetBytes := getWorkingSetV2(metrics.Memory)
+		oldStats = sandbox.Stats
```

Comment:
It doesn't look like `setContainerStats` is ever called with `isSandbox` set to true.  Can you remove the parameter and the codepath?